libnavui-app/src/main/java/com/mapbox/navigation/ui/app/internal/State.kt

```diff
@@ -0,0 +1,24 @@
+package com.mapbox.navigation.ui.app.internal
+
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.trip.session.LocationMatcherResult
+import com.mapbox.navigation.ui.app.internal.audioguidance.AudioGuidanceState
+import com.mapbox.navigation.ui.app.internal.camera.CameraState
+import com.mapbox.navigation.ui.app.internal.destination.Destination
+import com.mapbox.navigation.ui.app.internal.navigation.NavigationState
+import com.mapbox.navigation.ui.app.internal.routefetch.RoutesState
+import com.mapbox.navigation.ui.app.internal.tripsession.TripSessionStarterState
+
+/**
+ * Navigation state for internal use.
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+data class State(
```

Comment:
Is this supposed to be public API? I see that it's in the internal package but there's this annotation. Is it safe to use data classes for public API? I mean with all the additional methods like copy that may become binary incompatible when new params are added.

Reply:
Great catch :+1: The annotation is mostly redundant if the class is already in `.internal` package (with a goal to be shared internally between modules). When it comes to `data classes`, we avoid using them in public APIs unless we're extremely confident it won't need to be extended in the future, for most extensible use-cases, we fallback to a builder pattern which is automatically tested in [here](https://github.com/mapbox/mapbox-navigation-android/blob/v2.6.0-beta.2/libtesting-utils/src/main/java/com/mapbox/navigation/testing/BuilderTest.kt).

Since the goal of this class is to be internal, `data class` is fine.

Reply:
> My opinion is that it should opt-in, as any other actual internal class would.

Yeah, I'm on board. Internal implementations opt-in. If they are moved into public api they need to be changed to experimental if they are using experimental apis.